File: 002-Backend/LaPlata.API/LaPlata.Domain/Models/Despesas/Conta.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace LaPlata.Domain.Models
{
    public class Conta : AuditableModel
    {
        [Key]
        [Required]
        public int Id { get; set; }
        [Required]
        public string Descricao { get; set; }
        [Required]
        public int ValorInteiro { get; set; }
        public int? ValorCentavos { get; set; }
        [Required]
        public int DiaVencimento { get; set; }
        public virtual Categoria Categoria { get; set; }
        [Required]
        public int CategoriaId { get; set; }
        [JsonIgnore]
        public virtual List<PagamentoConta>? Pagamentos { get; set; }
    }
}

File: 003-Frontend/HomeTools.Front/src/app/interfaces/financas/Conta.ts
import { Categoria } from "../categoria";
import { PagamentoConta } from "./PagamentoConta";

export interface Conta {
  Id: number;
  Descricao: string;
  ValorInteiro: number;
  ValorCentavos: number;
  DiaVencimento: number;
  Categoria: Categoria;
  Pagamentos: PagamentoConta[];
  UltimoPagamento?: Date;
  Variavel: boolean;
  StatusId: number;
}
